upgrades/00_setup.md

```diff
@@ -0,0 +1,69 @@
+# Upgrading Your Leo Programs
+
+[ARC-0006](https://github.com/ProvableHQ/ARCs/discussions/94) proposes a number of changes to the AVM, which enable program upgradability.
+Different blockchain ecosystems take approaches to upgradability, with varying degrees of complexity, expressiveness, and security.
+The AVM is one point in this design space and Leo aims to provide a users with clean interface to develop and deploy programs, and to upgrade them in the future.
+
+## Resources for Program Upgradability
+
+- [ARC-0006](https://github.com/ProvableHQ/ARCs/discussions/94)
+- [snarkVM tracker](https://github.com/ProvableHQ/snarkVM/issues/2654)
+- [snarkOS PR](https://github.com/ProvableHQ/snarkOS/pull/3638)
+- [Leo PR](https://github.com/ProvableHQ/leo/pull/28593)
+
+The implementations for snarkVM, snarkOS, and Leo are all still in development.
+To start using these features, we'll need to set up a custom development environment.
+
+## Setting up snarkOS
+
+You will first need to clone the snarkOS repository and checkout the `feat/program-upgradability` branch.
+``` 
+> git clone https://github.com/ProvableHQ/snarkOS.git
+> cd snarkOS
+> git checkout feat/program-upgradability
```

Comment:
Can we make this a permalink so if someone wants to try out the feature they don't end up being on a broken commit while we put in the last fixes?